### ISSUE TITLE
CL Unit test fail with OSX M1 `accelerate` library

### ISSUE THREAD

**Issue by garrettwrong** [ComputationalCryoEM/ASPIRE-Python#838]
Just reporting I found the test below fails on my personal M1 mac when I use the `accelerate` backend. 

It only appears to occur with `accelerate`. Running with the current `develop` env, or the proposed `openblas`, `intel`, `default` backends all tests passed.  Considering this is only happening for one numerical library and only under float32 the issue is low priority (it may not be anything in ASPIRE). 

I'll try to repeat with the `accelerate` library on a different cpu at some point.

```
(aspire_accel) ➜  ASPIRE-Python git:(conda_backend_opts) ✗ pytest --lf
=========================================================================================== test session starts ============================================================================================
platform darwin -- Python 3.8.15, pytest-7.2.0, pluggy-1.0.0
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
rootdir: /Users/garrettwright/work_m1/ASPIRE-Python, configfile: tox.ini, testpaths: tests
plugins: random-order-1.1.0, xdist-3.1.0, cov-4.0.0, anyio-3.6.2
collected 598 items / 597 deselected / 1 selected                                                                                                                                                          
run-last-failure: rerun previous 1 failure (skipped 13 files)

tests/test_orient_symmetric.py F                                                                                                                                                                     [100%]

================================================================================================= FAILURES =================================================================================================
______________________________________________________________________________________ testCompleteThirdRow[float32] _______________________________________________________________________________________

dtype = <class 'numpy.float32'>

    @pytest.mark.parametrize("dtype", [np.float32, np.float64])
    def testCompleteThirdRow(dtype):
        # Complete third row that coincides with z-axis
        z = np.array([0, 0, 1], dtype=dtype)
        Rz = CLSymmetryC3C4._complete_third_row_to_rot(z)
    
        # Complete random third row.
        r3 = randn(3, seed=123).astype(dtype)
        r3 /= norm(r3)
        R = CLSymmetryC3C4._complete_third_row_to_rot(r3)
    
        # Assert that Rz is the identity matrix.
        assert np.allclose(Rz, np.eye(3, dtype=dtype))
    
        # Assert that R is orthogonal with determinant 1.
>       assert np.allclose(R @ R.T, np.eye(3, dtype=dtype))
E       AssertionError: assert False
E        +  where False = <function allclose at 0x7f7b59196790>((array([[-0.73934691, -0.6733247 ,  0.        ],\n       [-0.4715203 ,  0.51775478, -0.71386186],\n       [ 0.48066083, -0.52779156, -0.70028669]]) @ array([[-0.73934691, -0.4715203 ,  0.48066083],\n       [-0.6733247 ,  0.51775478, -0.52779156],\n       [ 0.        , -0.71386186, -0.70028669]])), array([[1., 0., 0.],\n       [0., 1., 0.],\n       [0., 0., 1.]], dtype=float32))
E        +    where <function allclose at 0x7f7b59196790> = np.allclose
E        +    and   array([[-0.73934691, -0.4715203 ,  0.48066083],\n       [-0.6733247 ,  0.51775478, -0.52779156],\n       [ 0.        , -0.71386186, -0.70028669]]) = array([[-0.73934691, -0.6733247 ,  0.        ],\n       [-0.4715203 ,  0.51775478, -0.71386186],\n       [ 0.48066083, -0.52779156, -0.70028669]]).T
E        +    and   array([[1., 0., 0.],\n       [0., 1., 0.],\n       [0., 0., 1.]], dtype=float32) = <function eye at 0x7f7b5923a700>(3, dtype=<class 'numpy.float32'>)
E        +      where <function eye at 0x7f7b5923a700> = np.eye

tests/test_orient_symmetric.py:401: AssertionError
========================================================================================= short test summary info ==========================================================================================
FAILED tests/test_orient_symmetric.py::testCompleteThirdRow[float32] - AssertionError: assert False
```

**Comment by garrettwrong** [comment]
Confirmed this also occurs with OSX on an i7 using the `accelerate` library.

**Comment by garrettwrong** [comment]
Noting this also occurs on GHA CI when attempting to use `accelerate`. It may be an issue with that GEMM call. Will try a few things.

**Comment by garrettwrong** [comment]
Was able to fix with `utest_tolerance` in #836 .
